### PR TITLE
Fixed #26643 -- Prevent unnecessary AlterModelManagers operations caused by the manager inheritance refactor.

### PR DESCRIPTION
**Pull request by loic**
https://code.djangoproject.com/ticket/26643

### CODE REVIEW

File: django/db/migrations/state.py [hunk 1]
```diff
@@ -444,28 +446,25 @@ def flatten_bases(model):
...
-            managers.append((force_text(default_manager.name), default_manager))
+                continue
+            managers.append((force_text(manager.name), new_manager))
 
-        for manager in model._meta.managers:
-            if manager.use_in_migrations and manager is not model._default_manager:
-                manager = copy.copy(manager)
-                manager._set_creation_counter()
-                managers.append((force_text(manager.name), manager))
+        # If all we have is a shimmed default manager called objects, we can just
```
Comment (timgraham) [inline django/db/migrations/state.py]: Lately I've noticed that avoiding "we" in comments often results in a simpler wording, e.g. `# Ignore a shimmed default manager called objects if it's the only one.`

File: tests/migrations/test_state.py [hunk 2]
```diff
@@ -191,6 +191,78 @@ class Meta:
         author_state = project_state.models['migrations', 'author']
         self.assertEqual(author_state.managers, [('authors', custom_manager)])
 
+    def test_custom_default_manager_named_objects_with_false_migration_flag(self):
+        """
+        When a manager is added with a name of 'objects', but it does not
```
Comment (timgraham) [inline tests/migrations/test_state.py]: chop comma

File: tests/migrations/test_state.py [hunk 3]
```diff
@@ -191,6 +191,78 @@ class Meta:
...
+        have use_in_migrations = True, no migration should be added to the
+        model state. Refs #26643.
```
Comment (timgraham) [inline tests/migrations/test_state.py]: state (#26643).

**Comment by timgraham** [comment]
Prevented*
Looks okay to me although I haven't worked in this area much.

**Comment by loic** [comment]
Thanks Tim, I incorporated all the suggestions.

**Pull request merged**